### DISCUSSION TITLE
ASP.NET Core VS Blazor

### DISCUSSION THREAD

**Discussion by ChuyGonzalezDev** [dotnet/aspnetcore#22091] [General]
We currently occupy ASP.NET Core, for frontend sites with JavaScript and JQuery with consumptions to API's Rest. We are evaluating switching to Blazor, however we are not so sure as we have created add-ons (JS) that help us develop forms faster and that is an advantage against our competition. Blazor how would you help create these same add-ons to make our developments faster?

**Comment by poke** [comment]
In my opinion, since you already appear to have made investments into JavaScript solutions, e.g. by creating custom libraries and components, the moving away from that and giving up your experience which currently is _“an advantage against [your] competition”_ does not sound like a great idea.

Blazor is really cool technology, and for teams that mostly come from .NET alone it is a great way to avoid JavaScript. But it’s not the silver bullet for everything that we will build in the future. So you should always consider whether it fits your situation.

If you right now have a working solution with JavaScript, stick to that and use your advantage. But if you and your team is interested, allow some space to play with it as it further evolves over the following months. After all, Blazor is only at the start of its young (supported) lifetime and there is still a lot ahead.

**Reply by ChuyGonzalezDev** [reply]
Thanks Patrick, it is a sensible idea to continue with what already works and gradually experiment with Blazor. Let's wait for future versions and we'll play with it. From the start, Blazor is doing great.

**Comment by MrOldham101** [comment]
Im not too sure I understand the wording of the question. To clarify, are you using a .NET Core back end that is ran completely separately to all your JS / JQuery front end sites and then using AJAX to interact with the .NET back end via a REST API built on the back end? Or are you using .Net for the front end as well using razor pages?
